client/src/components/CreatePost.tsx

```diff
@@ -1,180 +1,260 @@
-import React, { useState } from "react";
+import React, { useState } from 'react';
 
-import Grid from "@mui/material/Grid";
-import Box from "@mui/material/Box";
-import Container from "@mui/material/Container";
-import Button from "@mui/material/Button";
-import TextField from "@mui/material/TextField";
-import Divider from "@mui/material/Divider";
-import Typography from "@mui/material/Typography";
-import Tooltip from "@mui/material/Tooltip";
+import Grid from '@mui/material/Grid';
+import Box from '@mui/material/Box';
+import Container from '@mui/material/Container';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import Divider from '@mui/material/Divider';
+import Typography from '@mui/material/Typography';
+import Tooltip from '@mui/material/Tooltip';
+import Dialog from '@mui/material/Dialog';
+import { ArrowBack } from '@mui/icons-material';
+import PreviewPopUp from './PreviewPopUp';
+import Snackbar from '@mui/material/Snackbar';
 
-import PreviewPopUp from "./PreviewPopUp";
+import ServerApi from '../api/v1/index';
+
+const api = new ServerApi();
 
 function CreatePost() {
-  // create hooks for input fields
+  // hooks
   const [form, setForm] = useState({
-    title: "",
-    body: "",
-    file: "",
-    tags: "",
-    eventCapacity: "",
-    eventLocation: "",
+    title: '',
+    body: '',
+    file: '',
+    tags: '',
+    capacity: '',
+    location: '',
   });
+  const [openPopup, setOpenPopup] = useState(false); // for preview popup
+  const [isAlertOpen, showAlert] = useState(false); // for snackbar
+  const [alertMsg, setMsg] = useState('An error has occurred'); // for snackbar message
+  const [capacityError, setCapacityError] = useState(''); // for capacity input validation
+  const [isOpen, toggleDialog] = useState(false); // for create post dialog toggle
 
-  const [openPopup, setOpenPopup] = useState(false);
-
-  // handle functions
+  const closeDialog = () => {
```

Comment:
In closeDialog (before toggleDialog) empty the form (i.e set title, body, ..., to empty string)